openfisca_aotearoa/variables/rates_rebates.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# Import from openfisca-core the common python objects used to code the legislation in OpenFisca
+from openfisca_core.model_api import *
+# Import the entities specifically defined for this tax and benefit system
+from openfisca_aotearoa.entities import Propertee, Person
+from numpy import clip
+
+
+# This is a 'temporary' variable used specifically for the rates rebates
+# Due to the structure not yet being clear, this may need to be moved, renamed.
+class dependants(Variable):
+    value_type = int
+    entity = Person
+    definition_period = YEAR
+    label = u"Number of Persons classified as dependant on the entity Person for the purposes of rates rebates"
+
+
+class rates(Variable):
+    value_type = float
+    entity = Propertee
+    definition_period = YEAR
+
+
+class rates_rebate(Variable):
+    value_type = float
+    entity = Propertee
+    definition_period = YEAR
+    label = "Yearly rebate applied to housing rates. Defined on a calendar year, not a fiscal year."
+    reference = ""
+
+    def formula(properties, period, parameters):
+        income_threshold = parameters(period).benefits.rates_rebates.income_threshold
+        additional_per_dependant = parameters(period).benefits.rates_rebates.additional_per_dependant
+        initial_contribution = parameters(period).benefits.rates_rebates.initial_contribution
+        maximum_allowable = parameters(period).benefits.rates_rebates.maximum_allowable
+
+        # sum allowable income including all the dependants for property
+        allowable_income = (properties.sum(properties.members('dependants', period)) * additional_per_dependant) + income_threshold
+
+        # calculate the excess income based on the allowable income
+        excess_income = (properties.sum(properties.members('salary', period)) - allowable_income) / 8
```

Comment:
> This magic `8` reads very weird, are you sure it should not be a parameter? If not (i.e. the law states “divide by 8”, not “divide by something called ‘rate adjustment’ which has a value of 8 at the moment”), then it might be worth adding a little comment to make it explicit  🙂

Reply:
The law states `$1 for each $8 by which the ratepayer’s income for the preceding tax year exceeded $24,790`. There's no reasoning provided in the legislation - see:
http://www.legislation.govt.nz/act/public/1973/0005/67.0/DLM409673.html?search=sw_096be8ed8161c2a5_divide_25_se&p=1